hal_st/middlewares/ble_middleware/GapCentralSt.cpp

```diff
@@ -222,6 +250,15 @@ namespace hal
             });
     }
 
+    void GapCentralSt::HandleGapDirectConnectionProcedureEvent()
+    {
+        if (!initiatingStateTimer.Armed())
```

Comment:
Does not this need to be if (Armed)  ?
And don't we need to cancel the timer later?

Reply:
No, because this event happens in two scenarios:

1 - After the timeout occurs;
2 - After the function CancelConnect is called (then the timer is cancelled);

Reply:
Aligned: this check is useless, remove it.